library_generation/generate_composed_library.py

```diff
@@ -0,0 +1,141 @@
+"""
+This script allows generation of libraries that are composed of more than one
+service version. It is achieved by calling `generate_library.sh` without
+postprocessing for all service versions and then calling
+postprocess_library.sh at the end, once all libraries are ready.
+
+Prerequisites
+- Needs a folder named `output` in current working directory. This folder
+is automatically detected by `generate_library.sh` and this script ensures it
+contains the necessary folders and files, specifically:
+  - A "google" folder found in the googleapis/googleapis repository
+  - A "grafeas" folder found in the googleapis/googleapis repository
+Note: googleapis repo is found in https://github.com/googleapis/googleapis.
+"""
+
+import click
+import utilities as util
+import os
+import sys
+import subprocess
+import json
+from model.GenerationConfig import GenerationConfig
+from model.Library import Library
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+"""
+Main function in charge of generating libraries composed of more than one
+service or service version.
+Arguments
+ - config: a GenerationConfig object representing a parsed configuration
+ yaml
+ - library: a Library object contained inside config, passed here for
+   convenience and to prevent all libraries to be processed
+ - enable_postprocessing: true if postprocessing should be done on the generated
+   libraries
+ - repository_path: path to the repository where the generated files will be
+   sent. If not specified, it will be downloaded. The versions file will be inferred from this folder
+"""
+def generate_composed_library(
+    config: GenerationConfig,
+    library: Library,
+    enable_postprocessing: bool,
```

Comment:
Can we have a default value for `enable_postprocessing` and `repository_path`?

Reply:
I added a default value for `enable_postprocessing`

For `repository_path`, I added an explanation that it will be inferred
https://github.com/googleapis/sdk-platform-java/blob/f0b26ef41e15232efa164a6949e3ae3f8a69c7af/library_generation/generate_composed_library.py#L38-L40

The inference for the monorepo, for example, is done here:
https://github.com/googleapis/sdk-platform-java/blob/f0b26ef41e15232efa164a6949e3ae3f8a69c7af/library_generation/generate_composed_library.py#L80-L84